components/codeflare/codeflare.go

```diff
@@ -104,13 +104,18 @@ func (c *CodeFlare) ReconcileComponent(ctx context.Context, cli client.Client, l
 		return err
 	}
 	l.Info("apply manifests done")
+
+	// Wait for deployment available
+	if enabled {
+		// first check if the service is up, so prometheus won't fire alerts when it is just startup
+		if err := cluster.WaitForDeploymentAvailable(ctx, cli, ComponentName, dscispec.ApplicationsNamespace, 20, 2); err != nil {
+			return fmt.Errorf("deployment for %s is not ready to server: %w", ComponentName, err)
+		}
+	}
+
 	// CloudServiceMonitoring handling
 	if platform == cluster.ManagedRhods {
 		if enabled {
-			// first check if the service is up, so prometheus won't fire alerts when it is just startup
-			if err := cluster.WaitForDeploymentAvailable(ctx, cli, ComponentName, dscispec.ApplicationsNamespace, 20, 2); err != nil {
-				return fmt.Errorf("deployment for %s is not ready to server: %w", ComponentName, err)
-			}
 			l.Info("deployment is done, updating monitoring rules")
 		}
```

Comment:
Since we move this block outside, do we need this if again here? Same for all the other files as well